### ISSUE TITLE
Possible to make the plugin restartless

### ISSUE THREAD

**Issue by Chris2011** [jbisanti/Netbeans-UI-Editor-Plugin#11]
Hey,

Is it possible to make the plugin restartless or do you need some stuff, which is only available after restart?

Cheers

Chris

**Comment by jbisanti** [comment]
Unfortunately, some UI's are written in static way to where they initialize
certain colors, sizes, etc only once. In these cases, its typically done
the first time one of these components is instantiated and they never
change again, hence the need for a restart and why Java only officially
supports setting a UI at startup.

It might be possible to use reflection to work around these, but I haven't
had the time and don't foresee a time in the future when I might be able to
investigate this possibility. All that aside, thanks for using the plugin!

**Comment by Chris2011** [comment]
I really like it and often the problems are not fault of this plugin but it would be even more cool to make workarounds as most as possible, I really would like to jump in, but what I would like to be is a user of this plugin, which I sometimes do :) But maybe this great plugin could be a great enhancement of flexible change the laf of NetBeans, if you can donate the code to the NetBeans core.

So that a lot more users/developers can have a look there and can think about the stuff in NetBeans core. What do you think?

**Comment by jbisanti** [comment]
What's the process of donating to netbeans core?

**Comment by Chris2011** [comment]
You should create a jira ticket to what you want to achive like implementing a new ui editor to change your Laf maybe there must be fixed some stuff before, I dont know. Fork the netbeans code, make a branch whatever you want to work on this and if you think it is stable enough to created a module, you can make a PR. Maybe you/we should discuss this topic at the mailing list first.

**Comment by Chris2011** [comment]
But it is more easy nowadays to give code to the netbeans core or to the contrib repo (which is not part of the oracle -> apache donation yet but will be) as it was before. Maybe they have other solutions for that.
